lib/roby/interface/client.rb

```diff
@@ -101,7 +104,17 @@ def process_packet(m, *args)
                     e = args.first
                     raise e, e.message, (e.backtrace + caller)
                 elsif m == :reply
-                    yield args.first
+                    if !pending_async_calls.empty?
+                        current_call = pending_async_calls.shift
+                        if !current_call[:block].nil?
+                            current_call[:block].call args.first,
+                                                      current_call[:path],
```

Comment:
I don't think you need to forward the original call arguments to the block. If one wants to pass it to them, he/she can do it explicitely in their block (where context is available)

Reply:
You haven't handled the call failure codepath (:bad_call). I would say, just go the javascript way and pass the error as first argument, the return value as second.

Reply:
> I don't think you need to forward the original call arguments to the block. If one wants to pass it to them, he/she can do it explicitely in their block (where context is available)

Fair point.. This is a an old habit from C/C++ where callbacks usually have a "context/userdata" argument

> You haven't handled the call failure codepath (:bad_call). I would say, just go the javascript way and pass the error as first argument, the return value as second.

I wasn't sure `:bad_call` was a response of this calls also.. Will fix that.